cmd/prometheus-config-reloader/main.go

```diff
@@ -155,6 +159,19 @@ func main() {
 	}
 }
 
+func createHTTPClient() http.Client {
+	config := &tls.Config{
+		// TLS certificate verification is disabled by default
+		InsecureSkipVerify: true,
+	}
+
+	transport := &http.Transport{
```

Comment:
It would be good to clone `http.DefaultTransport` since this is what the reloader would use if no HTTP client is given.

```golang
        transport := (http.DefaultTransport.(*http.Transport)).Clone()
        transport.TLSClientConfig = config
```

Reply:
@simonpasquier I have gone ahead and done this alongside some changes to the transport timeout and max (idle) connections.
I based timeouts on my own tests on an AKS cluster, and local running Kind cluster. These values were within the margin (100ms margin).
Added comments to explain my changes and some can be removed/replaced once accepted.